packages/framework/src/client.ts

```diff
@@ -620,14 +621,20 @@ export class Client {
   }
 
   private async compileControls(templateControls: Record<string, unknown>, event: Event) {
-    const templateString = this.templateEngine.parse(JSON.stringify(templateControls));
+    try {
+      const templateString = this.templateEngine.parse(JSON.stringify(templateControls));
 
-    const compiledString = await this.templateEngine.render(templateString, {
-      ...(event.payload || event.data),
-      subscriber: event.subscriber,
-    });
+      const compiledString = await this.templateEngine.render(templateString, {
+        payload: event.payload || event.data,
+        subscriber: event.subscriber,
+        // Backwards compatibility, for allowing usage of variables without namespace (e.g. `{{name}}` instead of `{{payload.name}}`)
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do not leave false praise (which can actually be damaging). Do look for something to sincerely praise.  -->
👏 **praise:** Great comment!‏